backend/src/model/model.service.ts

```diff
@@ -0,0 +1,55 @@
+import { pipeline, env, PipelineType } from "@huggingface/transformers";
+import { ConfigService } from "../config/config.service";
+
+type Progress = { loaded: number; total: number };
+type ProgressCallback = (progress: Progress) => void;
+
+env.allowLocalModels = false;
+
+class PipelineSingleton {
+    static pipelines = new Map<string, any>();
+
+    static async getPipeline(task: string, model: string, progressCallback?: ProgressCallback) {
+        const pipelineOptions = progressCallback ? { progress_callback: progressCallback } : undefined;
+
+        const pipelineInstance = await pipeline(task as PipelineType, model, pipelineOptions);
+
+        if (!this.pipelines.has(model)) {
+            this.pipelines.set(model, pipelineInstance);
+        }
+        return this.pipelines.get(model);
+    }
+}
+
+export default async function loadAllChatsModels(progressCallback: ProgressCallback = () => {}) {
+    const configService = new ConfigService();
+    const chats = configService.get("chats");
+
+    for (const chatKey in chats) {
+        if (Object.prototype.hasOwnProperty.call(chats, chatKey)) {
```

Comment:
again, shouldn't provide unsafe config and read like this, so ugly 

do something like this, for example
```javascript
  const loadPromises = Object.entries(chats).map(async ([chatKey, chat]) => {
    const { model: modelName, task = "text-generation" } = chat;

    try {
      const pipelineInstance = await PipelineSingleton.getPipeline(
        task,
        modelName,
        progressCallback
      );
      
      console.log(`Model loaded successfully: ${modelName}`);
      loadedModels.set(chatKey, pipelineInstance);
    } catch (error) {
      console.error(`Failed to load model ${modelName}:`, error);
    }
  });

```